drivers/power/ltc7841/iio_ltc7841.c

```diff
@@ -0,0 +1,493 @@
+ * @param iio_desc - The iio device descriptor.
+ * @param init_param - The structure that contains the device initial parameters.
+ * @return 0 in case of success, an error code otherwise.
+ */
+int ltc7841_iio_init(struct ltc7841_iio_desc **iio_desc,
+		     struct ltc7841_iio_desc_init_param *init_param)
+{
+	struct ltc7841_iio_desc *descriptor;
+	int ret;
+
+	if (!init_param || !init_param->ltc7841_init_param)
+		return -EINVAL;
+
+	descriptor = no_os_calloc(1, sizeof(*descriptor));
+	if (!descriptor)
+		return -ENOMEM;
+
+	ret = ltc7841_init(&descriptor->ltc7841_desc,
+			   init_param->ltc7841_init_param);
+	if (ret)
+		goto dev_err;
+
+	descriptor->iio_dev = &ltc7841_iio_dev;
+
+	*iio_desc = descriptor;
+
+	return 0;
+
+dev_err:
+	no_os_free(descriptor);
+
+	return ret;
+}
+
+/**
+ * @brief Free resources allocated by the init function.
+ * @param desc - The iio device descriptor.
+ * @return 0 in case of success, an error code otherwise.
+ */
+int ltc7841_iio_remove(struct ltc7841_iio_desc *desc)
+{
+	if (!desc)
+		return -ENODEV;
+
+	no_os_free(desc->iio_dev->channels);
+	ltc7841_remove(desc->ltc7841_desc);
+	no_os_free(desc);
+
+	return 0;
+}
+
+/**
+ * @brief Handles the read request for raw attribute.
+ * @param dev     - The iio device structure.
+ * @param buf	  - Command buffer to be filled with requested data.
+ * @param len     - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv    - Command attribute id.
+ * @return ret    - Result of the reading procedure.
+ * 		    In case of success, the size of the read data is returned.
+*/
+static int ltc7841_iio_read_raw(void *dev, char *buf, uint32_t len,
+				const struct iio_ch_info *channel,
+				intptr_t priv)
+{
+	struct ltc7841_iio_desc *iio_ltc7841 = dev;
+	struct ltc7841_desc *ltc7841 = iio_ltc7841->ltc7841_desc;
+	int ret;
+	uint8_t read_value[2];
+	uint16_t temp = 0;
+	int32_t value;
+
+	switch (channel->address) {
+	case LTC7841_IIO_VIN_CHAN: {
+		ret = ltc7841_reg_read(ltc7841, (uint8_t)LTC7841_READ_VIN, read_value);
+		if (ret)
+			break;
+		temp = no_os_get_unaligned_le16(read_value);
+		break;
+	}
+	case LTC7841_IIO_VOUT_CHAN: {
+		ret = ltc7841_reg_read(ltc7841, (uint8_t)LTC7841_READ_VOUT, read_value);
+		if (ret)
+			break;
+		temp = no_os_get_unaligned_le16(read_value);
+		break;
+	}
+	case LTC7841_IIO_IIN_CHAN:
+		ret = ltc7841_reg_read(ltc7841, (uint8_t)LTC7841_READ_IIN, read_value);
+		if (ret)
+			break;
+		temp = no_os_get_unaligned_le16(read_value);
+		break;
+	case LTC7841_IIO_IOUT_CHAN:
+		ret = ltc7841_reg_read(ltc7841, (uint8_t)LTC7841_READ_IOUT, read_value);
+		if (ret)
+			break;
+		temp = no_os_get_unaligned_le16(read_value);
+		break;
+	case LTC7841_IIO_TEMP_CHAN:
+		ret = ltc7841_reg_read(ltc7841, (uint8_t)LTC7841_READ_TEMPERATURE_1,
+				       read_value);
+		if (ret)
+			break;
+		temp = no_os_get_unaligned_le16(read_value);
+		break;
+	default:
+		return -EINVAL;
```

Comment:
You might want to be consistent with your brackets in case statements. But, I don't see a need to add one.

Reply:
brackets use in the case statements are still in consistent.